aspnet/fundamentals/localization.rst

```diff
@@ -0,0 +1,337 @@
+
+.. code-block:: c#
+	:emphasize-lines: 6-9
+
+	public void Configure(IApplicationBuilder app, IStringLocalizer<Startup> SR)
+	{
+		// Add the platform handler to the request pipeline.
+		app.UseIISPlatformHandler();
+
+		var options = new RequestLocalizationOptions();
+		options.RequestCultureProviders.Clear();
+		options.RequestCultureProviders.Add(new QueryStringRequestCultureProvider());
+		app.UseRequestLocalization(options);
+
+		...
+	}
+
+Using the QueryStringRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When you instantiate a ``RequestLocalizationOptions`` object, the ``QueryStringRequestCultureProvider`` is added to the list of providers as the first provider to be used in attempting to determine the culture. The ``QueryStringRequestCultureProvider`` looks for the existence of two keys in the query string: "culture" and "ui-culture". If the provider finds a value for "culture", but not for "ui-culture", the UI culture is set to the culture value. Likewise, if the provider finds a value for "ui-culture", but not for culture, the culture is set to the UI culture value. Therefore, when using the ``QueryStringRequestCultureProvider``, all of the following URIs will result in a culture and UI culture of Spanish.
+
+::
+
+	// All of these URIs will result in a culture and UI culture of Spanish.
+	http://<yourApp>/?culture=es-ES&ui-culture=es-ES
+	http://<yourApp>/?culture=es-ES
+	http://<yourApp>/?ui-culture=es-ES
+
+Note that while the query string keys are case insensitive, the locale value (es-ES, in this example) is case sensitive. Therefore, the following URIs will not result in a found culture as the locale is  not correctly cased:
+
+::
+
+	// The QueryStringRequestCultureProvider will not locate a valid culture
+	// based on this URI as the correct locale is es-ES (case sensitive).
+	http://<yourApp>/?ui-culture=es-es
+
+One last note about the ``QueryStringRequestLocalizationProvider`` is that if either culture or ui-culture is not valid, the result is null - meaning that a culture was not located. The next provider in the list (if there is one) is then given the opportunity to determine the culture.
+
+Using the CookieRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``CookieRequestCultureProvider`` looks for the existence of a cookie whose name is represented by ``CookieRequestCultureProvider.CookieName``. When you instantiate a ``CookieRequestCultureProvider`` object, the ``CookieRequestCultureProvider.CookieName`` value is set to the value of ``CookieRequestCultureProvider.DefaultCookieName``. The following example illustrates an example of allowing the user to create (and clear) the cookie.
+
+.. code-block:: c#
```

Comment:
I think this example is unnecessary and can be removed. The CookieRequestCultureProvider is added by default.